HunyuanImage2.1: Implement Hunyuan Mixed APG and Fix Refiner text template

KimbingNg opened the pull request:
HunyuanImage2.1: Implement Hunyuan Mixed APG and Fix Refiner text template

A workflow example:
[HunyuanImage-2.1.json](https://github.com/user-attachments/files/22348070/HunyuanImage-2.1.json)

cc @comfyanonymous

Kosinkadink commented on comfy_extras/nodes_hunyuan.py:
```diff
@@ -23,6 +25,220 @@ def encode(self, clip, bert, mt5xl):
+            adaptive_projected_guidance_momentum=ocr_momentum
+        )
+
+        current_step = {"step": 0}
```
Gotcha, in that case, what you are looking for is making an ```OUTER_SAMPLE``` wrapper, an [example of which is done for the easycache_sample_wrapper in EasyCache nodes for core](https://github.com/comfyanonymous/ComfyUI/blob/4f1f26ac6c11b803bbc83cb347178e2f9b5e421b/comfy_extras/nodes_easycache.py#L144 ).

IS_CHANGED does not work how you would expect - returning True does NOT make it run every time, and is also not the way to make sure the code runs every time the sampler is ran; it is separate from the sampling process entirely. The return value of IS_CHANGED is like a 'fingerprint' - if this fingerprint is different than the last time the node instance was run, then the execution function of the node is permitted to run. If the fingerprint is the same, the node will NOT run and the cached value will be used instead. Returning True from IS_CHANGED means the node will run ONLY ONCE the whole time it is on the graph. In V3 schema, this function got renamed to ```fingerprint_inputs``` to become more clear in its effects.

What you'll want to do is make sure the dictionary you want to access gets reinitialized in the ```OUTER_SAMPLE``` wrapper. The function that ```OUTER_SAMPLE``` wraps runs every time sampling is initiated in ComfyUI, so it's the perfect place to reset the step value back to zero. This function also encapsulates sampling before CFG, so it won't get screwed up by the amount of conditioning being passed in.

comfyui-wiki commented:
## Original
<img width="2048" height="2048" alt="original" src="https://github.com/user-attachments/assets/00c5265d-19f1-4273-afa1-083cc201651e" />

## Updated
<img width="2048" height="2048" alt="updated" src="https://github.com/user-attachments/assets/deb8bc4a-293c-4fd1-b3a1-a513e741bb0d" />

Prompt 
```
At a local jazz club, three musicians are captured mid-performance on a small stage. In the foreground, a saxophonist with salt-and-pepper hair takes the lead, captured mid-howl with his saxophone to his mouth; he wears a deep purple velvet blazer over a white collared shirt. Behind him, a younger woman with a vibrant afro plucks the strings of a double bass, her body swaying to the music in a sparkling, blue dress. To the side, a man in his mid-30s, wearing a fedora, bangs on the drums, his arms covered in intricate tattoos. The club is dimly lit with warm, ambient light, with spotlights highlighting the performers on stage. In the softly focused background, a crowd of onlookers watches the performance, their faces illuminated by the gentle glow of the room. Photography style. Medium-close up shot.
```

@Kosinkadink  This is the comparison between our implementation and this PR.

KimbingNg commented:
@Kosinkadink We adjusted the code to use percentage-based setting in the latest commit and will extract the refiner part separately into another PR.

@wpdong0727  Thanks for your valuable feedback, we have fixed this bug in the latest commit.
